Documentation overhaul

AnnaWiniwarter opened the pull request:
This PR contains a complete overhaul of the readthedocs documentation page.
I tested the compiled website, but it would be a good idea if you could check that. Otherwise, I am not aware of which formatting requirements you have. 
I ran invoke black, which lead to the zilien.py file being updated, not sure why that happened. 
Sorry for the not-so-professional PR here. 

Also, once merged, will the readthedocs page be updated automatically?

ScottSoren commented on docs/source/getting_started/tutorials/ecms_tutorial.rst:
```diff
@@ -0,0 +1,1054 @@
+            None,
+            [0, 20],
+        ],  # [left, right] y-axes
+        unit=["nmol/s", "pmol/s/cm^2"],  # [left, right] y-axes
```
It's fine demo of the code, but wierd, I think, to have H2 and O2 in different units in this case. Maybe save different left and right units for a time where one axis plots a carrier gas?

ScottSoren commented on docs/source/getting_started/tutorials/ecms_tutorial.rst:
```diff
@@ -0,0 +1,1054 @@
+    # grab the iongauge pressure
+    ig1 = meas_d.grab("Iongauge value [mbar]")
+    
+    # create a figure with 3 panels
```
Soon, you could use `ixdat.plotters.base_mpl_plotter.MPLPlotter.new_three_panel_axes()` for this :) 
https://github.com/ixdat/ixdat/blob/785bf2e0cab0ec8069fb8bcbf25104fc3d02dfca/src/ixdat/plotters/base_mpl_plotter.py#L91

ScottSoren commented on docs/source/getting_started/tutorials/ecms_tutorial.rst:
```diff
@@ -0,0 +1,1054 @@
+.. image:: EC-MS_ixdat_tutorial_files/EC-MS_ixdat_tutorial_61_0.png
+
+
+EC-MS plot with a third with system parameters added
```
Nice! :)

ScottSoren commented on docs/source/getting_started/tutorials/index.rst:
```diff
@@ -0,0 +1,77 @@
+
+The basics of importing and plotting from each reader are demonstrated in
+the **development_scripts/reader_testers** folder of the repository:
+https://github.com/ixdat/ixdat/tree/user_ready/development_scripts/reader_testers
```
This link should be updated to:
https://github.com/ixdat/ixdat/tree/main/development_scripts/reader_demonstrators

ScottSoren commented on docs/source/getting_started/tutorials/sec_tutorials.rst:
```diff
@@ -0,0 +1,196 @@
+It shows delta optical density calculation and both calculation and plotting of the full 2-D data field and
+cross sections (i.e. spectra and wavelength-vs-time).
+
+The example data is not yet publically available, thus there is no code execution shown here. 
```
It may be publically available now! The article is under review and available in pre-print. https://www.researchsquare.com/article/rs-2605628/v1
@Caiwu-L and I will fix this up :)

ScottSoren replied:
as this is not urgent, we'll wait until that article is accepted :)

ScottSoren commented on docs/source/index.rst:
```diff
@@ -34,30 +34,19 @@ Or rather than exporting, you can take advantage of ``ixdat``'s powerful analysi
+If you have any feedback, comments or questions, find out how to contact the ixdat team here: :ref:`questions`. 
+This documentation, like ``ixdat`` itself, is a work in progress and we appreciate any feedback or requests.
 
 .. toctree::
```
Nice! Huge improvement in the organization. 

Hey, I wonder if we should also have a page linking to some of our slide decks on ixdat?

AnnaWiniwarter replied:
I think that would be a great idea, but maybe not for this PR, so we can get this out asap?

ScottSoren commented on docs/source/introduction.rst:
```diff
@@ -5,19 +5,26 @@ Introduction
+  :width: 620
+  :alt: Design: pluggability
+
+Get started today with one of the tutorials available under :ref:`tutorials`. If you're curious about the details of ``ixdat``'s structure and the available readers, plotters and exporters, check out the section :ref:`diving_deeper`. 
```
Nice!

ScottSoren reviewed (changes requested):
Hi @AnnaWiniwarter ,

Thank you for doing this! This looks fantastic! I cloned your repository to build the documentations and read it all in .html while commenting.
Not all the comments require action, but some request some small changes, should be quick. I think it's high priority to merge this in, so anything that isn't quick should be done in a later PR.  Yeah, I think it'll automatically build to readthedocs when we merge the PR.

The most commented parts are the tutorials. I take it you used a tool to convert jupyter notebooks to .rst? What tool did you use? 
A problem is that the output blocks are indestinguishable from the code blocks. So hopefully we can modify the tool a bit to make them distinct.

AnnaWiniwarter commented on docs/source/diving_deeper/structure_docs/measurement.rst:
```diff
@@ -67,6 +38,9 @@ A measurement is basically a wrapper around a collection of ``data_series`` (see
 
 There are several ways of interracting with a measurement's ``data_series``:
 
+- Most TechniqueMeasurements provide attribute-style access to essential DataSeries and
```
To be honest, I don't really remember. But in general, I think it makes more sense to start with the more defined, specific properties, and then introduce the more general way. Otherwise, why even the first option, if the best thing is to use grab() for everything anyway.

AnnaWiniwarter replied:
I will leave it as it is now

AnnaWiniwarter commented on docs/source/diving_deeper/structure_docs/technique_docs/mass_spec.rst:
```diff
@@ -11,7 +10,8 @@ tracked in time.
 The main TechniqueMeasurement class for MID data is the ``MSMeasurement``.
 
-Classes dealing with spectra are under development.
+The position of spectra in ixdat is not completely set yet. Currently, they exist as ``Spectrum`` base
```
Cool! Can I ask you to update this paragraph/write a paragraph that I can add here instead (if you want this update to be part of this PR already, that is).

AnnaWiniwarter commented on docs/source/getting_started/cheat_sheet.rst:
```diff
@@ -0,0 +1,77 @@
+.. _cheat_sheet:
+
+===========
+Cheat sheet
```
yeah me too, but then I ran out of time, so I thought better like this than nothing at all....

AnnaWiniwarter commented on docs/source/getting_started/tutorials/advanced_tutorials.rst:
It is!! There's a link to it in the tutorials/index.rst file!

AnnaWiniwarter commented on docs/source/getting_started/tutorials/ec_tutorials.rst:
This raises a valid point though: The way I converted jupyter notebooks into rst is very manual, and for sure won't update if you change anything in the notebook. I didn't think about this initially, but I can see that that might become an issue eventually. Regardless, I think it's really, really important that we have the executed notebook on the documentation page, because having to run a notebook just creates a way, way too high barrier for the not-so-programming affine future ixdat users!

AnnaWiniwarter replied:
Maybe we should discuss this elsewhere...

AnnaWiniwarter commented:
@ScottSoren I have now addressed all the issues you had, except for the ones that are directly related to the EC-MS tutorial, as I feel this is really scope creeping here. I understand the things you mentioned (in some cases I have already created issues to fix the underlying problems in the ixdat methods), but I wonder if it's absolutely necessary to deal with all of these before merging this PR? I promise I will take care of them afterwards :) 
I will look into how to fix the issue of visualizing the difference between text and code blocks though...

ScottSoren reviewed (approved):
Hey @AnnaWiniwarter , 
This looks good! As you point out, tutorials requires some discussion and will be addressed in another PR. We've left unresolved all the comments on the tutorial rst's here for reference at that time, and I've just made an Issue. 
Github seems to have confused itself, so I'm seeing all the commits from #101 in the diff's, but hopefully the merge to main will go smooth. There's some small linting error which I'll fix after merging and make sure it passes tests.
Otherwise, amazing, important improvement here that will make ixdat a lot more usable and promotable. Hopefully readthedocs updates automatically when it merges (I will check). Thanks for the PR!
